pints/_mcmc/_adaptive_covariance_localised.py

```diff
@@ -0,0 +1,245 @@
+
+        # Return proposed point
+        return self._proposed
+
+    def _initialise(self):
+        """
+        See :meth: `AdaptiveCovarianceMCMC._initialise()`.
+        """
+        super(AdaptiveCovarianceLocalisedMCMC, self)._initialise()
+        self._localised = True
+        self._mixture_components = 3
+        
+        # Initialise weights
+        self._w = np.repeat(1.0 / self._mixture_components,
+                            self._mixture_components)
+
+        # Create lists of randomised means and covariance matrices
+        epsilon_mu = np.random.normal(0.01 * self._mu, 1, size=(self._mixture_components,
+                                      self._dimension))
+        epsilon_sigma_v = np.random.normal(np.zeros(self._dimension),
+                                           0.01,
+                                           size=(self._mixture_components,
+                                                 self._dimension))
+        self._epsilon_sigma = []
+        for i in range(self._mixture_components):
+            dsigm = np.reshape(epsilon_sigma_v[i], (self._dimension, 1))
+            self._epsilon_sigma.append(np.dot(dsigm, dsigm.T))
+        a_temp = np.copy(self._mu)
+        a_temp_sigma = np.copy(self._sigma)
+        self._mu = []
+        self._sigma = []
+        for i in range(self._mixture_components):
+            self._mu.append(a_temp + epsilon_mu[i])
+            self._sigma.append(a_temp_sigma + self._epsilon_sigma[i])
+
+        # Initialise lambda vector
+        self._log_lambda = np.zeros(self._mixture_components)
+        
+        # Initialise running expected acceptance probabilities
+        self._alpha_l = np.zeros(self._mixture_components)
+        
+        # Initialise log_q_l
+        self._log_q_l = np.log(np.repeat(1.0 / self._mixture_components,
+                                         self._mixture_components))
+
+    def set_mixture_components(self, mixture_components):
+        """
+        Sets the number of Gaussian mixture components
+        to use for proposals
+        """
+        if mixture_components < 2:
+            raise ValueError('Number of mixture components ' +
+                             'should exceed 1.')
+        if not isinstance(mixture_components, int):
+            raise ValueError('Number of mixture components ' +
+                             'should be an integer.')
+        self._mixture_components = mixture_components
+
+    def tell(self, fx):
+        """ See :meth:`pints.AdaptiveCovarianceMCMC.tell()`. """
+        super(AdaptiveCovarianceLocalisedMCMC, self).tell(fx)
+        
+        # Return new point for chain
+        return self._current
+        
+    def _fit_gaussian_mixture(self):
+        """
+        Fits a Gaussian mixture distribution by updating
+        componentwise the means, covariance matrices,
+        weights and lamdas (eq. 36 and 37 in Andrieu &
+        Thoms 2008)
+        """
+        self._update_log_q()
```

Comment:
Do we need these to all be in separate functions? Are they called in more than one place? If not, then shouldn't be functions!

Reply:
Also thinking, perhaps we should merge update_mu and update_sigma into a single function, then perhaps we don't need the split into local/non-local in tell.
Would also be great if we could update the names of all these things to explain what they do in words, `update_arbitrary_symbol` isn't very descriptive (to me - perhaps I don't know the methods well enough?)